app/src/main/java/com/example/bigowlapp/repository/Repository.java

```diff
@@ -152,9 +149,9 @@ public Repository(FirebaseFirestore mFirebaseFirestore, CollectionReference coll
     // Fetching a list of Documents
     //===========================================================================================
```

Comment:
Since this PR contains lots of refactors, I have a small suggestion.

I think it would be fine if you removed all duplicated code by extract method.

Reply:
I'll leave it for the cleanup branch we are working on.